allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -89,15 +127,42 @@ def tokens_to_indices(
                     f" for the following token: {token.text}"
                 )
 
-        # The mask has 1 for real tokens and 0 for padding tokens. Only real
-        # tokens are attended to.
-        attention_mask = [1] * len(indices)
+        if self._intra_word_tokenization:
+            # self._intra_word_tokenize() does not insert special tokens, so we need to do it here
+            indices = self._underlying_tokenizer.build_inputs_with_special_tokens(indices)
+            offsets = [
+                (start + self._num_added_start_tokens, end + self._num_added_start_tokens)
+                for start, end in offsets
+            ]
 
-        return {"token_ids": indices, "mask": attention_mask}
+        output = {"token_ids": indices, "mask": mask}
+        if self._intra_word_tokenization:
+            output["wordpiece_mask"] = [1] * len(indices)
+            output["offsets"] = offsets  # type: ignore
```

Comment:
Why do you need a `# type: ignore` here?  You probably just need a more permissive type annotation on line 138 - mypy probably inferred a too-specific type from the inline dictionary instantiation.

Reply:
Oh, nevermind, is this because of my cheating with `IndexedTokenList`?

Reply:
I think mypy just infers the dict value type to be `List[int]`, either from the creation statement or using the return type, and is unhappy when I insert a value with type `List[Tuple[int, int]]`. I'm not sure if it has something to do with `IndexedTokenList`, although I wonder if it'd be nicer to be a little more generic with `IndexedTokenList` and allow it to be `Dict[str, List[Any]]`?

Reply:
Yeah, if it doesn't break other things, feel free to loosen the definition of `IndexedTokenList`, or use a `TypeVar`, or something.